homeassistant/components/mqtt/client.py

```diff
@@ -573,12 +686,16 @@ async def _async_perform_subscriptions(
         def _process_client_subscriptions() -> list[tuple[int, int]]:
             """Initiate all subscriptions on the MQTT client and return the results."""
             subscribe_result_list = []
-            for topic, qos in subscriptions:
+            for topic, qos in subscriptions.items():
                 result, mid = self._mqttc.subscribe(topic, qos)
                 subscribe_result_list.append((result, mid))
                 _LOGGER.debug("Subscribing to %s, mid: %s, qos: %s", topic, mid, qos)
             return subscribe_result_list
 
+        async with self._pending_subscriptions_lock:
+            subscriptions = copy.deepcopy(self._pending_subscriptions)
```

Comment:
Why do we deepcopy, the dict's values are immutable?

Reply:
We are creating a local copy to be able to process the update outside the lock